### ISSUE TITLE
Docker development build high CPU usage while idle

### ISSUE THREAD

**Issue by ahmedashraf443** [stackblitz-labs/bolt.diy#566]
### Describe the bug

I’ve deployed the bolt.new production build using Docker, and while it generally runs without issues, I’ve observed unusually high CPU usage when the container is idle. Additionally, the following error occasionally appears in the logs, potentially pointing to socket-related issues:

2024-12-04 20:55:13 TypeError: terminated
    at Fetch.onAborted (node:internal/deps/undici/undici:10822:53)
    at Fetch.emit (node:events:519:28)
    at Fetch.emit (node:domain:488:12)
    at Fetch.terminate (node:internal/deps/undici/undici:9980:14)
    ...
2024-12-04 20:55:13   [cause]: SocketError: other side closed
    at Socket.<anonymous> (node:internal/deps/undici/undici:6040:28)
    at Socket.emit (node:events:531:35)
    at endReadableNT (node:internal/streams/readable:1696:12)
    ...

2024-12-06 18:23:50 [warn] Data fetching is changing to a single fetch in React Router v7
2024-12-06 18:23:50 ┃ You can use the `v3_singleFetch` future flag to opt-in early.
2024-12-06 18:23:50 ┃ -> https://remix.run/docs/en/2.13.1/start/future-flags#v3_singleFetch
2024-12-06 18:23:50 ┗

2024-12-06 18:08:57 > bolt@ dev /app
2024-12-06 18:08:57 > remix vite:dev "--host" "0.0.0.0"
2024-12-06 18:08:57 
2024-12-06 18:09:02   ➜  Local:   http://localhost:5173/
2024-12-06 18:09:02   ➜  Network: http://172.21.0.2:5173/
2024-12-06 18:18:28  ELIFECYCLE  Command failed.

### Link to the Bolt URL that caused the error

Docker development build

### Steps to reproduce

1-Clone the repository or follow the setup instructions for the production build using Docker.
2-Start the production container and let it run idle.
3-Observe the Docker logs; the error appears intermittently.
4-Monitor CPU usage of the container during idle periods.

### Expected behavior

The container should maintain low CPU usage when idle and not produce any socket termination errors.

### Screen Recording / Screenshot

The CPU usage of the container is unusually high during idle periods.
The following error appears intermittently in the logs:
TypeError: terminated
[cause]: SocketError: other side closed
![image](https://github.com/user-attachments/assets/39842b80-68dc-47d9-8477-8fe03aee196a)
![1](https://github.com/user-attachments/assets/c90e09fa-b2e3-4814-9ed9-27e920bd8085)

### Platform

OS: Windows 11

**Comment by leex279** [comment]
@ahmedashraf443 does this problem still occure for you in the newest release (or even main branch)?

**Comment by jwillmer** [comment]
Yes, I tried bolt yesterday and I have the same issue. I tried starting bolt in production, hoping that the required CPU overhead is development/debug related, but run into:

```
Attaching to app-prod-1
app-prod-1  |
app-prod-1  | > bolt@0.0.6 dockerstart /app
app-prod-1  | > bindings=$(./bindings.sh) && wrangler pages dev ./build/client $bindings --ip 0.0.0.0 --port 5173 --no-show-interactive-dev-session
app-prod-1  |
app-prod-1  | sh: 1: ./bindings.sh: not found
app-prod-1  |  ELIFECYCLE  Command failed.
app-prod-1 exited with code 1
```

So right now I'm also stuck with 50% CPU usage. That is annoying because it also slows down the AI generation a lot for me.

**Comment by aliasfoxkde** [comment]
Generally the error `ELIFECYCLE  Command failed.` means your NPM packages are jacked up. Try running a `npm cache verify` to fix it (or do a complete purge and re-install). Also, the utilization is 68.74 out of 1200% (12 CPUs available), so affectively 6% of total resources. Maybe bump up the resources for docker if this is an issue?

Another note is that Bolt.diy in "Dev" mode is running a full Remix server, so it's actually doing a bit under the hood. You could use the pre-built version available through `npm install -g bolt.diy & bolt.diy` instead which should be much lighter. Lastly, keep in mind that docker on Mac/Windows is virtualized, so it's not given full resource allocation and it adds overhead. Docker is native only to Linux.

**Comment by jwillmer** [comment]
Yes I know that but it is definetly using to much process power. I virtualize a lot and never get such numbers. I have run it now Natively via npm and have 1-2% instead of 50-60%
